scripts/submit-analytics.js

```diff
@@ -0,0 +1,201 @@
+//
+////////////////////////////////////////////////////////////////////////////
+
+const commandLineArgs = require("command-line-args");
+const fs = require("fs");
+
+let doLog; // placeholder for logger function
+
+class Webhook {
+  /**
+   * Path and credentials required to submit analytics through the webhook (production mode).
+   */
+  constructor() {
+    this.urlPrefix =
+      "https://webhooks.mongodb-realm.com/api/client/v2.0/app/realmsdkmetrics-zmhtm/service/metric_webhook/incoming_webhook/metric?ip=1&data=";
+  }
+
+  /**
+   * Constructs the full URL that will submit analytics to the webhook.
+   * @param  {Object} payload Information that will be submitted through the webhook.
+   * @returns {string} Complete analytics submission URL
+   */
+  buildRequest(payload) {
+    const request = this.urlPrefix + Buffer.from(JSON.stringify(payload.webHook), "utf8").toString("base64");
+    return request;
+  }
+}
+
+/**
+ * Generate a hash value of data
+ * @param {*} data
+ * @returns SHA256 of data
+ */
+function sha256(data) {
+  let hash = require("crypto").createHash("sha256");
+  hash.update(data);
+  return hash.digest("hex");
+}
+
+/**
+ * Collect analytics data from the runtime system
+ * @param {Object} context The app's package.json parsed as an object
+ * @returns {Object} Analytics payload
+ */
+async function fetchPlatformData(context) {
+  const os = require("os");
+  const { machineId } = require("node-machine-id");
+
+  let identifier = await machineId();
+  if (!identifier) {
+    identifier = sha256("unknown");
+  }
+
+  let framework = "node.js";
+  let frameworkVersion = process.version;
+  let jsEngine = "v8";
+
+  if (context.dependencies && context.dependencies["react-native"]) {
+    framework = "react-native";
+    frameworkVersion = context.dependencies["react-native"];
+    try {
+      const podfile = fs.readFileSync("../../ios/Podfile", "utf8");
+      if (podfile.includes("hermes_enabled => true")) {
+        jsEngine = "hermes";
+      } else {
+        jsEngine = "jsc";
+      }
+    } catch (err) {
+      doLog(`Cannot read ios/Podfile: ${err}`);
+      jsEngine = "unknown";
+    }
+  }
+  if (context.dependencies && context.dependencies["electron"]) {
+    framework = "electron";
+    frameworkVersion = context.dependencies["electron"];
+  }
+
+  const payloads = {
+    webHook: {
+      event: "install",
+      properties: {
+        token: "aab85907a13e1ff44a95be539d9942a9",
+        "JS Analytics Version": 2,
+        distinct_id: identifier,
+        "Anonymized Machine Identifier": identifier,
+        "Anonymized Application ID": sha256(__dirname),
+        Binding: "javascript",
+        Version: context.version,
+        Language: "javascript",
+        Framework: framework,
+        "Framework Version": frameworkVersion,
+        "JavaScript Engine": jsEngine,
+        "Host OS Type": os.platform(),
+        "Host OS Version": os.release(),
+        "Node.js version": process.version,
```

Comment:
Is it deliberate that this is lowercase `version` and everything else is `Version`?